### PR TITLE
[SPARK-15552][SQL] Remove unnecessary private[sql] methods in SparkSession

### PR DESCRIPTION
**Pull request by rxin**
## What changes were proposed in this pull request?

SparkSession has a list of unnecessary private[sql] methods. These methods cause some trouble because private[sql] doesn't apply in Java. In the cases that they are easy to remove, we can simply remove them. This patch does that.

As part of this pull request, I also replaced a bunch of protected[sql] with private[sql], to tighten up visibility.
## How was this patch tested?

Updated test cases to reflect the changes.

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala [hunk 1]
```diff
@@ -178,12 +180,14 @@ class SparkSession private(
...
 
   /**
+   * :: Experimental ::
    * Returns a [[ContinuousQueryManager]] that allows managing all the
    * [[org.apache.spark.sql.ContinuousQuery ContinuousQueries]] active on `this`.
    *
    * @group basic
    * @since 2.0.0
    */
+  @Experimental
```
Comment (rxin) [inline sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala]: this is a "bug" fix

**Comment by rxin** [comment]
cc @cloud-fan @andrewor14

**Comment by andrewor14** [comment]
retest this please

**Comment by andrewor14** [comment]
m2.0

**Pull request closed without merging**